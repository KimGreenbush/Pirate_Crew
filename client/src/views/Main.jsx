import React, {useEffect} from "react"
import Pirates from "../components/Pirates"
import { Link } from "@reach/router"
import axios from "axios"

const Main = ({pirates, setPirates}) => {
    useEffect(() => {
        axios.get("http://localhost:9001/api/pirate")
            .then(res => setPirates(res.data))
            .catch(err => console.log("Error: ", err))
    }, [])

    return (
        <main>
            <header>
                <h1>Pirate Crew</h1>
                <img src="https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Ftcibythelake.com%2Fwp-content%2Fuploads%2F2018%2F06%2Fpirate-ship-150x150.png&f=1&nofb=1" alt="pirate ship" />
            </header>
            <Link to="/pirate/new"><button>Add Pirate</button></Link>
            <Pirates pirates={pirates} setPirates={setPirates} />
        </main>
    )
}

export default Main