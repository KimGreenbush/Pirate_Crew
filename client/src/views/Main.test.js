/**
 * @jest-environment jsdom
 */

import React from "react";
import Main from "./Main";
import renderer from "react-test-renderer";

test("renders Main without crashing", () => {
	const component = renderer.create(<Main />);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
