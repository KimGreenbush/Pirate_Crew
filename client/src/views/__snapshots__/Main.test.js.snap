// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Main without crashing 1`] = `
<main>
  <header>
    <h1>
      Pirate Crew
    </h1>
    <img
      alt="pirate ship"
      src="https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Ftcibythelake.com%2Fwp-content%2Fuploads%2F2018%2F06%2Fpirate-ship-150x150.png&f=1&nofb=1"
    />
  </header>
  <a
    href="/pirate/new"
    onClick={[Function]}
  >
    <button>
      Add Pirate
    </button>
  </a>
  <div
    className="pirates"
  >
    
  </div>
</main>
`;
